script/dwc_mapping.Rmd

```diff
@@ -0,0 +1,397 @@
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = TRUE)
+```
+
+Load libraries:
+
+```{r message = FALSE}
+library(tidyverse)      # To do data science
+library(tidylog)        # To provide feedback on dplyr functions
+library(magrittr)       # To use %<>% pipes
+library(here)           # To find files
+library(janitor)        # To clean input data
+library(readxl)         # To read Excel files
+library(rgbif)          # To use GBIF services
+library(uuid)           # To generate UUID's
+```
+
+# Read source data
+
+
+
+```{r}
+input_data <- read_delim(here::here("output", "gbif_stierkikker_vangsten.csv"), ",")
+
+
+```
+
+
+Preview data:
+
+```{r}
+input_data %>% head()
+```
+
+
+
+# Preprocessing
+
+## Tidy data
+
+Clean data somewhat:
+
+```{r}
+input_data %<>% 
+  remove_empty("rows") %<>%
+  clean_names()
+```
+
+
+
+
+### fix unique ID issue
+
+```{r}
+input_data <- mutate_all(input_data, as.character())
+```
+
+
+# Darwin Core mapping
+
+Map the data to [Darwin Core Occurrence](http://rs.gbif.org/core/dwc_occurrence_2015-07-02.xml).
+
+```{r}
+occurrence <- input_data
+```
+
+### type
+
+```{r}
+occurrence %<>% mutate(dwc_type = "Event")
+```
+
+### language
+
+```{r}
+occurrence %<>% mutate(dwc_language = "en")
+```
+
+### license
+
+```{r}
+occurrence %<>% mutate(dwc_license = "http://creativecommons.org/publicdomain/zero/1.0/")
+```
+
+### rightsHolder
+
+```{r}
+occurrence %<>% mutate(dwc_rightsHolder = "INBO")
+```
+
+### accessRights
+
+```{r}
+occurrence %<>% mutate(dwc_accessRights = "http://www.inbo.be/en/norms-for-data-use") 
+```
+
+### datasetID
+
+```{r}
+occurrence %<>% mutate(dwc_datasetID = "")
+```
+
+### institutionCode
+
+
+### datasetName
+
+```{r}
+occurrence %<>% mutate(dwc_datasetName = "Invasive Bullfrog occurrences in Flanders")
+```
+
+### basisOfRecord
+
+```{r}
+occurrence %<>% mutate(dwc_basisOfRecord = "HumanObservation")
+```
+
+### occurrenceID
+
+
+```{r}
+occurrence %<>% mutate(dwc_occurrenceID = occurrence_id)
+```
+
+
+
+
+### recordedBy
+
+```{r}
+occurrence %<>% mutate(dwc_recordedBy = 'INBO team FIS')
+```
+
+### individualCount
+
+```{r}
+#mutate(occurrence$mm)as.numeric()
+  
+occurrence %<>% mutate(dwc_individualCount = individual_count)
+
+```
+
+
+
+
+### occurrenceStatus
+
+```{r}
+occurrence %<>% mutate(dwc_occurrenceStatus = case_when(
+  dwc_individualCount > 0 ~ "present",
+  dwc_individualCount == 0 ~ "absent"
+))
+```
+
+### eventID
+
+
+```{r}
+occurrence %<>% mutate(dwc_eventID = str_replace_all(event_id, " ",""))
+```
+
+### eventDate
+
+```{r}
+occurrence %<>% mutate(dwc_eventDate = datum)
+```
+
+
+
+
+### habitat
+
+```{r}
+occurrence %<>% mutate(dwc_habitat = recode(type
+                                            , 'Stilstaand' = 'freshwater; stagnant'))
+```
+
+### samplingProtocol
+
+```{r}
+occurrence %<>% mutate(dwc_samplingProtocol = recode(actie
+                                                     ,"Afvangst" = "catch, fike"
```

Comment:
Hergebruik van een andere mapping ?

Reply:
yes, but will be useful when you decide to use other traps. ;)